packages/provider/src/app/provider.ts

```diff
@@ -52,6 +52,30 @@ export class OptimismProvider extends JsonRpcProvider {
       return tx
     }
 
+    const blockFormat = this.formatter.block.bind(this.formatter)
+    this.formatter.block = (block) => {
+      const b = blockFormat(block)
+      b.stateRoot = block.stateRoot
+      return b
+    }
+
+    const blockWithTransactions = this.formatter.blockWithTransactions.bind(
+      this.formatter
+    )
+    this.formatter.blockWithTransactions = (block) => {
+      const b = blockWithTransactions(block)
+      b.stateRoot = block.stateRoot
+      for (let i = 0; i < b.transactions.length; i++) {
+        b.transactions[i].meta = {
```

Comment:
I think we should still remove this `meta` field and place the properties directly on the tx or update geth to return a meta field. I lean towards the first, its less work and safer because its less nesting. Things will be better when the null propagation operator gets into js